### ISSUE TITLE
Handle S3 OperationAborted due to conflicting conditions

### ISSUE THREAD

**Issue by jaypipes** [aws/aws-controllers-k8s#188]
If you attempt to create a Bucket with the same name as a Bucket that you recently deleted, you might run into this:

```
OperationAborted: A conflicting conditional operation is currently in progress against this resource.
```

This is due to S3 queuing the delete operation and the need to have a unique name of the bucket for the account. Attempting to create the Bucket with the same name violates the uniqueness constraint for the still-being-deleted previous Bucket.

We should properly handle this error and trigger a delay in the Retry operation in the reconciler.

**Comment by jaypipes** [comment]
The solution here is twofold:

* Add some logic into the `ExceptionsConfig` struct in the generatorConfig to indicate certain error codes should be caught and simply cause a delayed retry of the original request.
* Change the template in `crd_sdk.go.tpl` to handle these kinds of exceptions.

**Comment by DanStough** [comment]
Is this the right direction with the changes to the generatorConfig? Wanted to be clear that "error codes" doesn't mean HTTP codes in this case - there are multiple 409's you can get when create a bucket and I'm assuming you wouldn't want to requeue for `BucketAlreadyExists`.
```
resources:
  Bucket:
  ...
    exceptions:
      requeue:
        - OperationAborted
```

**Comment by jaypipes** [comment]
> Is this the right direction with the changes to the generatorConfig? Wanted to be clear that "error codes" doesn't mean HTTP codes in this case - there are multiple 409's you can get when create a bucket and I'm assuming you wouldn't want to requeue for `BucketAlreadyExists`.
> 
> ```
> resources:
>   Bucket:
>   ...
>     exceptions:
>       requeue:
>         - OperationAborted
> ```

Yeah, I realized way too late that S3 Bucket names are globally unique, and that `OperationAborted: A conflicting condition operation...` is returned when trying to **Create** an S3 Bucket with the same name as one currently being deleted.

This will need to be handled quite carefully; I'm still thinking about ways we can instruct the code generator about these types of weirdnesses.
